src/workerd/io/io-context.h

```diff
@@ -1059,14 +1064,24 @@ jsg::Promise<IoContext::MaybeIoOwn<addIoOwn, T>> IoContext::awaitIoImpl(
             }
           }
           KJ_CASE_ONEOF(exception, kj::Exception) {
-            resolver.reject(lock, kj::mv(exception));
+            resolver.resolve(kj::mv(exception));
```

Comment:
Here, as well, a comment explaining why we're resolving with an exception would be helpful for future generations.